### PR TITLE
Send partition values instead of index name in OpenTableRequest

### PR DESCRIPTION
**Pull request by mfussenegger**
Helps https://github.com/crate/crate/pull/16964
To support table swap or rename without having to close and open shards
it's necessary that a index has no relation name association.

That means index names will be replaced by UUIDs and to identify
partitions we need to keep holding onto relationName + partition values
or the index uuid.

### CODE REVIEW

File: server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import org.elasticsearch.Version;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
+
+
+public class OpenTableRequest extends AcknowledgedRequest<OpenTableRequest> {
```
Comment (seut) [inline server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java:37]: It feels a bit strange that now a `OpenTableRequest` is also used on the `CloseTableAction`. Worth adding an empty `CloseTableRequest` extending on this? Or we call it just `TableRequest`?

**Review by seut — APPROVED** [review]
Lgtm otherwise

File: server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    public List<String> partitionValues() {
+        return partitionValues;
+    }
+
+    public OpenTableRequest(StreamInput in) throws IOException {
+        super(in);
+        if (in.getVersion().before(Version.V_5_10_0)) {
+            relation = new RelationName(in);
+            String partitionIndexName = in.readOptionalString();
+            partitionValues = PartitionName.fromIndexOrTemplate(partitionIndexName).values();
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java:66]: We can have NPE here, `PartitionName.fromIndexOrTemplate` doesn't work with null values whereas `partitionIndexName` can be null. 

I think we need a test for that

File: server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java:37]: Where is it used? I don't see it in `TransportCloseTable`

File: server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/execution/ddl/tables/OpenTableRequest.java]: Good catch. Added a fixup

**Pull request merged**